### PR TITLE
doc: Defines migration guide for 1.23.0 release

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Capturing latest changes using changelog as reference.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .changelog/2825.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-```release-note:note
```
Comment (AgustinBettati) [inline .changelog/2825.txt:1]: pass by fix of these changelog entries so they are placed within enhancement section

**Review by EspenAlbert — APPROVED** [review]
LGTM

File: .changelog/2825.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+```release-note:enhancement
 resource/mongodbatlas_advanced_cluster: Adds new `advanced_configuration.0.default_max_time_ms` attribute
```
Comment (lantoli) [inline .changelog/2825.txt]: nit: no need to say "new"

File: docs/guides/1.23.0-upgrade-guide.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+- You can now pin a cluster's FCV before upgrading in Atlas, which enables you to revert MongoDB to the previous version after upgrading without experiencing version compatibility issues. This is supported in `mongodbatlas_advanced_cluster` and `mongodbatlas_cluster` using `pinned_fcv`. To learn more, please review either [mongodbatlas_advanced_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) or [mongodbatlas_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster) resource documentation.
+
+- You can now configure the default time limit in milliseconds for individual read operations to complete in `mongodbatlas_advanced_cluster` using `advanced_configuration.0.default_max_time_ms`. To learn more, please review [mongodbatlas_advanced_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) resource documentation.
+
+
```
Comment (lantoli) [inline docs/guides/1.23.0-upgrade-guide.md:16]: I would keep the "Terraform MongoDB Atlas modules" section as in previous guides

File: docs/guides/1.23.0-upgrade-guide.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+page_title: "Upgrade Guide 1.23.0"
+---
+
+# MongoDB Atlas Provider 1.23.0: Upgrade and Information Guide
+
+The Terraform MongoDB Atlas Provider version 1.23.0 has a number of new and exciting features.
+
+## New Resources, Data Sources, and Features
+
+- You can now leverage auto-scaling to scale the instance size of each individual shard independently for sharded and geo-sharded clusters defined with `mongodbatlas_advanced_cluster`. To learn more, see the [Advanced Cluster New Sharding Configurations - Use Auto-Scaling Per Shard](advanced-cluster-new-sharding-schema#use-auto-scaling-per-shard).
```
Comment (lantoli) [inline docs/guides/1.23.0-upgrade-guide.md]: nit: should we better use an absolute url so it always goes to the latest version (once 1.23.0 is not the latest version) as in the links below?

**Review by lantoli — APPROVED** [review]

File: docs/guides/1.23.0-upgrade-guide.md (hunk 3 above)
Comment (AgustinBettati) [reply docs/guides/1.23.0-upgrade-guide.md:16]: no strong opinion on including in all guides moving forward, was their any conversation around this for previous guides?

Comment (lantoli) [reply docs/guides/1.23.0-upgrade-guide.md:16]: it was commented in the previous guides PR. maybe you can confirm with Zuhair

Comment (AgustinBettati) [reply docs/guides/1.23.0-upgrade-guide.md:16]: synced with @Zuhairahmed, we can include this section when there are updates for Modules

File: docs/guides/1.23.0-upgrade-guide.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+The Terraform MongoDB Atlas Provider version 1.23.0 has a number of new and exciting features.
+
+## New Resources, Data Sources, and Features
+
+- You can now leverage auto-scaling to scale the instance size of each individual shard independently for sharded and geo-sharded clusters defined with `mongodbatlas_advanced_cluster`. To learn more, see the [Advanced Cluster New Sharding Configurations - Use Auto-Scaling Per Shard](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema#use-auto-scaling-per-shard).
+
+- You can now pin a cluster's FCV before upgrading in Atlas, which enables you to revert MongoDB to the previous version after upgrading without experiencing version compatibility issues. This is supported in `mongodbatlas_advanced_cluster` and `mongodbatlas_cluster` using `pinned_fcv`. To learn more, please review either [mongodbatlas_advanced_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) or [mongodbatlas_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster) resource documentation.
```
Comment (kanchana-mongodb) [inline docs/guides/1.23.0-upgrade-guide.md]:
```suggestion
- You can now pin a cluster's FCV before upgrading in Atlas, which enables you to revert MongoDB to the previous version after upgrading without experiencing version compatibility issues. This is supported in `mongodbatlas_advanced_cluster` and `mongodbatlas_cluster` using `pinned_fcv`. To learn more, see either [mongodbatlas_advanced_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) or [mongodbatlas_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster) resource documentation.
```

File: docs/guides/1.23.0-upgrade-guide.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+- You can now configure the default time limit in milliseconds for individual read operations to complete in `mongodbatlas_advanced_cluster` using `advanced_configuration.0.default_max_time_ms`. To learn more, please review [mongodbatlas_advanced_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) resource documentation.
```
Comment (kanchana-mongodb) [inline docs/guides/1.23.0-upgrade-guide.md]:
```suggestion
- You can now configure the default time limit in milliseconds for individual read operations to complete in `mongodbatlas_advanced_cluster` using `advanced_configuration.0.default_max_time_ms`. To learn more, see [mongodbatlas_advanced_cluster](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) resource documentation.
```

**Review by kanchana-mongodb — APPROVED** [review]
LGTM % non-blocking nits

**Pull request merged**